packages/typespec-client-generator-core/src/public-utils.ts

```diff
@@ -520,12 +531,19 @@ function buildNameFromContextPaths(
   // 2. build name
   let createName: string = "";
   for (let j = lastNonAnonymousModelNodeIndex; j < contextPath.length; j++) {
-    if (!contextPath[j]?.type?.name) {
+    const currContextPathType = contextPath[j]?.type;
+    if (
+      currContextPathType?.kind === "String" ||
+      currContextPathType?.kind === "Number" ||
+      currContextPathType?.kind === "Boolean"
+    ) {
+      createName = `${createName}${contextPath[j].displayName}${capitalCase(String(currContextPathType.value))}`;
```

Comment:
A minor concern on naming, whether it be `${createName}${contextPath[j].displayName}}` or `${createName}${contextPath[j].displayName}${capitalCase(String(currContextPathType.value))}`

A real case in storage blob
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/storage/data-plane/Microsoft.BlobStorage/stable/2021-12-02/blob.json#L6237-L6248
to 
https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/SyncCopyStatusType.java

However, I guess if we really need a specified name in TypeSpec, we can do e.g.
```
enum SyncCopyStatusType {
  Success: "success"
}

model .. {
  `x-ms-copy-status`?: SyncCopyStatusType.Success
}
```